### ISSUE TITLE
Type Declarations to extend GlideRecord for Data Table Attributes

### ISSUE THREAD

**Issue by RaynorUE** [RaynorUE/snich#138]
Long story short, found a nifty way that we might be able to get something like... 
``` 
var inc = new GlideRecord('incident');
inc.addQuery('');
inc.query();
inc.number; <--- to get that to be part of the intellisense, and also set as a GlideElement type..
```

However, it would require doing something like this in front of your variable:
```
/**
 * @type {incident}   <--- where this would the table name
 */
var test = new myTest('hello');
```

Then the next question is... do we load all non application file tables for an app scope when "loading all app files" and generating the necessary type definition files at that time? 

And/Or probably best to make it it's own command(s)?
- Load Type Definitions for Application (single-select)
- Load Type Definitions for Table  (multi-select)

- [x] Add methods to pull table column data down, column name and type
- [x] Add method to generate TS Definition file from table data
- [x] Store TS Definition table inside the vscode hidden folders, store each table in a unique file (so we don't have to manage grouping)
- [x] Add an command pallete action, with appropriate questions, make one of those "launch user guide (turn off in settings) and a setting to control it showing or not
- [x] Add another question, about turning off the prmpt for user guide video for them.... disable both questions... 
- [x] Add to the "Load all application files" looking for sys_dictionary entries in that app scope (grouped by table?) and pre-load them accordingly..

**Comment by RaynorUE** [comment]
Ended up handling this approach by creating a "typescript definition" table in ServiceNow and end up adding that to the table config to sync down with a .d.ts file extension.

Then on the SN Side we added code to auto-generate these for various things such as table definitions. However, this also allows to save / create any form / structure of type script definitions we want include custom object types. 

In short, it turns out you can add your own .d.ts files and they'll get read appropriately!

**Comment by RaynorUE** [comment]
Would be good to revisit this to perform it automatically... but need to consider the volume of tables it could pull... 
do we do some form of auto evaluation based on the tables their current app interacts with? Do we make it it's own command since tables don't change to often?
